packages/expo-audio/android/src/main/java/expo/modules/audio/service/AudioRecordingService.kt

```diff
@@ -0,0 +1,201 @@
+      PendingIntent.getActivity(
+        this,
+        0,
+        appIntent,
+        PendingIntent.FLAG_UPDATE_CURRENT or PendingIntent.FLAG_IMMUTABLE
+      )
+    }
+
+    return NotificationCompat.Builder(this, CHANNEL_ID)
+      .setContentTitle("Recording audio")
+      .setContentText("Tap to return to app")
+      .setSmallIcon(android.R.drawable.ic_btn_speak_now)
+      .setOngoing(true)
+      .setContentIntent(contentIntent)
+      .addAction(
+        android.R.drawable.ic_delete,
+        "Stop",
+        stopPendingIntent
+      )
+      .setCategory(NotificationCompat.CATEGORY_SERVICE)
+      .build()
+  }
+
+  private fun startForegroundWithNotification() {
+    val notification = buildNotification()
+
+    try {
+      if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+        startForeground(
+          notificationId,
+          notification,
+          ServiceInfo.FOREGROUND_SERVICE_TYPE_MICROPHONE
+        )
+      } else {
+        startForeground(notificationId, notification)
+      }
+    } catch (_: Exception) {
+    }
+  }
+
+  fun registerRecorder(recorder: AudioRecorder) {
+    activeRecorders.removeAll { it.get() == null }
+    activeRecorders.add(WeakReference(recorder))
+
+    if (activeRecorders.size == 1) {
+      startForegroundWithNotification()
+    }
+  }
+
+  fun unregisterRecorder(recorder: AudioRecorder) {
+    activeRecorders.removeAll { it.get() == recorder || it.get() == null }
+
+    if (activeRecorders.isEmpty()) {
+      stopSelf()
+    }
+  }
+
+  private fun stopRecordingAndService() {
+    activeRecorders.forEach { weakRef ->
+      weakRef.get()?.stopRecording()
+    }
+    activeRecorders.clear()
+    stopSelf()
+  }
+
+  override fun onBind(intent: Intent?): IBinder {
+    return binder
+  }
+
+  override fun onDestroy() {
+    super.onDestroy()
+    instance = null
+    activeRecorders.clear()
+  }
+
+  companion object {
+    private const val CHANNEL_ID = "expo_audio_recording_channel"
+    private const val NOTIFICATION_ID = 2001
+    private const val ACTION_START_RECORDING = "expo.modules.audio.action.START_RECORDING"
+    private const val ACTION_STOP_RECORDING = "expo.modules.audio.action.STOP_RECORDING"
+
+    @Volatile
+    private var instance: AudioRecordingService? = null
+    private val pendingRecorders = mutableSetOf<AudioRecorder>()
+
+    fun getInstance(): AudioRecordingService? = instance
+
+    fun startService(context: Context, recorder: AudioRecorder) {
+      val intent = Intent(context, AudioRecordingService::class.java).apply {
+        action = ACTION_START_RECORDING
+      }
+
+      val service = getInstance()
+      if (service != null) {
+        service.registerRecorder(recorder)
+      } else {
+        synchronized(pendingRecorders) {
+          pendingRecorders.add(recorder)
+        }
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+          context.startForegroundService(intent)
```

Comment:
Do we need to provide the Android 14+ fix here where we set the service type when starting?